File: frontend/src/domain/entities/InspectionResult.ts
import type { InspectionVerdict } from '@/domain/types/inspection'

export class InspectionResult {
  readonly id: string
  readonly inspectionItemId: string
  readonly verdict: InspectionVerdict
  readonly evidenceIds: string[]
  readonly createdAt: string // ISO 8601 UTC形式（ローカルで発行）
  readonly createdBy: string
  readonly note?: string

  constructor(
    id: string,
    inspectionItemId: string,
    verdict: InspectionVerdict,
    evidenceIds: string[],
    createdAt: string,
    createdBy: string,
    note?: string
  ) {
    this.id = id
    this.inspectionItemId = inspectionItemId
    this.verdict = verdict
    this.evidenceIds = evidenceIds
    this.createdAt = createdAt
    this.createdBy = createdBy
    this.note = note
  }
}

File: backend/src/main/kotlin/com/localbridge/backend/entity/InspectionResult.kt
package com.localbridge.backend.entity

import jakarta.persistence.*
import java.time.Instant

@Entity
@Table(name = "inspection_results")
data class InspectionResult(
    @Id
    @Column(length = 36)
    val id: String,

    @Column(name = "inspection_item_id", nullable = false, length = 36)
    val inspectionItemId: String,

    @Column(nullable = false, length = 50)
    @Enumerated(EnumType.STRING)
    val verdict: InspectionVerdict,

    @Column(columnDefinition = "TEXT")
    val note: String? = null,

    @Column(name = "evidence_ids", columnDefinition = "TEXT")
    val evidenceIds: String, // JSON array stored as string

    @Column(name = "created_by", nullable = false, length = 255)
    val createdBy: String,

    @Column(name = "created_at", nullable = false)
    val createdAt: Instant = Instant.now()
)

enum class InspectionVerdict {
    OK,
    NG,
    N_A
}
